smdebug/core/access_layer/utils.py

```diff
@@ -34,12 +34,9 @@ def training_has_ended(trial_prefix):
             f"The end of training job file will not be written for jobs running under SageMaker."
         )
         return
-    try:
-        check_dir_exists(trial_prefix)
-        # if path does not exist, then we don't need to write a file
-    except RuntimeError:
-        # dir exists
-        pass
+
+    check_dir_exists(trial_prefix)
```

Comment:
What is the purpose of this line, since the return value appears to be ignored and no error is raised?  As far as I can tell, this `check_dir_exists(trial_prefix)` check doesn't do anything, correct?  Should it maybe be removed, or made part of an `if` clause?

Reply:
I was personally debating that as well. `check_dir_exists` performs a call to S3, and would raise an exception if the exception wasn't just `NoSuchBucket` (e.g. invalid credentials, etc.). This would in turn skip the below logic for the writer, which could have been part of the desired impact from  the previous implementation (where it basically said `if dir exists, do nothing, if dir doesn't exist, do nothing, otherwise, catch some other exceptions`).

So ultimately, it's an early termination of the function in case there's some S3 config issues, which isn't completely necessary, but helpful

Reply:
Ok, that makes sense.  I'd maybe just add a comment explaining that then, for the sake of readability.

Reply:
added a short comment